src/ploneintranet/workspace/browser/tiles/sidebar.py

```diff
@@ -127,22 +136,60 @@ def users(self):
     def existing_users(self):
         return self.workspace().existing_users()
 
-    def __call__(self):
+    def execute_batch_function(self):
         form = self.request.form
         ws = self.workspace()
-        user_id = form.get('user_id')
-        if user_id:
-            if not self.can_manage_workspace():
+        user_ids = form.get('user_id')
+        if isinstance(user_ids, basestring):
+            user_ids = user_ids.split(',')
+        batch_function = form.get('batch-function')
+        if user_ids:
+            if not self.can_manage_roster():
                 msg = _(u'You do not have permission to change the workspace '
                         u'policy')
                 raise Unauthorized(msg)
             else:
-                IWorkspace(ws).add_to_team(user=user_id)
-                api.portal.show_message(
-                    _(u'Member added'),
-                    self.request,
-                    'success',
-                )
+                if batch_function == 'add':
+                    for user_id in user_ids:
+                        IWorkspace(ws).add_to_team(user=user_id)
+                    api.portal.show_message(
+                        _(u'Member(s) added'),
+                        self.request,
+                        'success',
+                    )
+                elif batch_function == 'remove':
+                    for user_id in user_ids:
+                        IWorkspace(ws).remove_from_team(user=user_id)
+                    api.portal.show_message(
+                        _(u'Member(s) removed'),
+                        self.request,
+                        'success',
+                    )
+                elif batch_function == 'role':
+                    role = self.request.get('role')
+                    default_role = ws.participant_policy.title()
+                    groups = [default_role]
+                    if role:
+                        groups.append(role)
```

Comment:
I get your point, visually. However the wording of the interactions makes it clear this is about assigning roles, not assigning *extra* roles. Probably fixable by replacing the default group with the new one, instead of setting both.